File: Inventory/Models/Product.cs
public class Product
{
    public Datum[] data { get; set; }
    public Meta meta { get; set; }
}

public class Meta
{
    public Pagination pagination { get; set; }
}

public class Pagination
{
    public int start { get; set; }
    public int limit { get; set; }
    public int total { get; set; }
}

public class Datum
{
    public string productId { get; set; }
    public string upc { get; set; }
    public Aislelocation[] aisleLocations { get; set; }
    public string brand { get; set; }
    public string[] categories { get; set; }
    public string countryOrigin { get; set; }
    public string description { get; set; }
    public Image[] images { get; set; }
    public Item[] items { get; set; }
    public Iteminformation itemInformation { get; set; }
    public Temperature temperature { get; set; }
}

public class Iteminformation
{
    public string depth { get; set; }
    public string height { get; set; }
    public string width { get; set; }
}

public class Temperature
{
    public string indicator { get; set; }
    public bool heatSensitive { get; set; }
}

public class Aislelocation
{
    public string bayNumber { get; set; }
    public string description { get; set; }
    public string number { get; set; }
    public string numberOfFacings { get; set; }
    public string side { get; set; }
    public string shelfNumber { get; set; }
    public string shelfPositionInBay { get; set; }
}

public class Image
{
    public string perspective { get; set; }
    public Size[] sizes { get; set; }
    public bool featured { get; set; }
}

public class Size
{
    public string size { get; set; }
    public string url { get; set; }
}

public class Item
{
    public string itemId { get; set; }
    public bool favorite { get; set; }
    public Fulfillment fulfillment { get; set; }
    public Price price { get; set; }
    public string size { get; set; }
    public string soldBy { get; set; }
}

public class Fulfillment
{
    public bool curbside { get; set; }
    public bool delivery { get; set; }
    public bool inStore { get; set; }
    public bool shipToHome { get; set; }
}

public class Price
{
    public float regular { get; set; }
    public float promo { get; set; }
}

File: Inventory/ClientApp/src/app/product.ts
export interface Product {
    data: Datum[];
    meta: Meta;
}

export interface Meta {
    pagination: Pagination;
}

export interface Pagination {
    start: number;
    limit: number;
    total: number;
}

export interface Datum {
    productId: string;
    upc: string;
    aisleLocation: Aislelocation[];
    brand: string
    categories: string[];
    countryOrigin: string;
    description: string;
    images: Image[];
    items: Item[];
    itemInformation: Iteminformation;
    temperature: Temperature;
}

export interface Iteminformation {
    depth: string;
    height: string;
    width: string;
}

export interface Temperature {
    indicator: string;
    heatSensitive: boolean;
}

export interface Aislelocation {
    bayNumber: string;
    description: string;
    number: string;
    numberOfFacings: string;
    side: string;
    shelfNumber: string;
    shelfPositionInBay: string;
}

export interface Image {
    perspective: string;
    sizes: Size[];
    featured: boolean;
}

export interface Size {
    size: string;
    url: string;
}

export interface Item {
    itemId: string;
    favorite: boolean;
    fulfillment: Fulfillment;
    price: Price;
    size: string;
    soldBy: string;
    inventory: ProductInv; // This is where our info goes. It links up with the itemId
}

export interface Fulfillment {
    curbside: boolean;
    delivery: boolean;
    inStore: boolean;
    shipToHome: boolean;
}

export interface Price {
    regular: number;
    promot: number;
}

export interface ProductInv { // This is a new object we are inserting into the Item interface.
    id: number;
    productName: string;
    onHand: number;
    sales: number;
    itemId: string;
}